### PR TITLE
glyr 1.0.8 (new formula) and optional dependency in abcde

### PR DESCRIPTION
**Pull request by rwstauner**
I'm not sure what the guidelines/preferences are for optional dependencies, but if your abcde config file specifies `getalbumart` in `ACTIONS` `abcde` will refuse to run if it cannot find `glyrc`.

### CODE REVIEW

File: Library/Formula/glyr.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
+class Glyr < Formula
+  desc "Music related metadata searchengine with command-line interface and C API"
+  homepage "https://github.com/sahib/glyr"
+
```
Comment (bfontaine) [inline Library/Formula/glyr.rb]: Please remove this empty line.

File: Library/Formula/glyr.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
+class Glyr < Formula
+  desc "Music related metadata searchengine with command-line interface and C API"
```
Comment (bfontaine) [inline Library/Formula/glyr.rb]: “Search engine” is in two words.

File: Library/Formula/glyr.rb [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  depends_on "glib"
+  depends_on "gettext"
+
+  def install
+    system "cmake", ".", *std_cmake_args
+    system "make", "install"
+  end
+
+  test do
+    system "#{bin}/glyrc", "--version"
```
Comment (bfontaine) [inline Library/Formula/glyr.rb]: Can’t we write a test that does something more substantial than e.g. `--version` or `--help`?

Comment (rwstauner) [inline Library/Formula/glyr.rb]: To be honest I don't know.  I'm not very familiar with the tool, but it looks as though every function downloads something from the internet and I wasn't sure if that was desired.
I'd be happy to try something.

File: Library/Formula/glyr.rb (hunk 2 above)
Comment (rwstauner) [inline Library/Formula/glyr.rb]: Sorry, copied it straight from the home page.  The homebrew audit noticed `commandline` but not that one :)

**Comment by rwstauner** [comment]
If this library is too obscure that's fine.  I hadn't heard of it myself until trying to use this new `abcde` feature and got an error that this thing was missing.
I considered just making a tap for it, but thought i'd start with this since `abcde` does expect it under certain configurations.

File: Library/Formula/glyr.rb [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+  depends_on "glib"
+  depends_on "gettext"
+
+  def install
+    system "cmake", ".", *std_cmake_args
+    system "make", "install"
+  end
+
+  test do
+    (out = testpath/"cover.txt").write("x")
```
Comment (bfontaine) [inline Library/Formula/glyr.rb]: Do you really need to create the file before running the command below?

Comment (rwstauner) [inline Library/Formula/glyr.rb]: No, I suppose not.  It's just an old habit of initializing it so you can be certain it changed during the test.
Is `testpath` a new directory every time?

Comment (bfontaine) [inline Library/Formula/glyr.rb]: Yes.

Comment (rwstauner) [inline Library/Formula/glyr.rb]: Removed and rebased. Thanks!

**Comment by rwstauner** [comment]
This is my first new formula so I appreciate you taking the time to comment.

File: Library/Formula/glyr.rb [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+
+  def install
+    system "cmake", ".", *std_cmake_args
+    system "make", "install"
+  end
+
+  test do
+    out = testpath/"cover.txt"
+    system "#{bin}/glyrc", "cover", "-D", "--artist", "Beatles", "--album", "Please Please Me", "-w", out
+    assert File.read(out).match(%r{^https?://})
```
Comment (bfontaine) [inline Library/Formula/glyr.rb]: Please use `assert_match %r{^https?://}, out.read`

Comment (rwstauner) [inline Library/Formula/glyr.rb]: sweet, thanks.

**Comment by dunn** [comment]
Notable enough, and important to have dependencies of our existing formula. Merged in b7b470ab926637b144f7632b64a730906a6a986b, thanks!

**Comment by rwstauner** [comment]
Thanks!

**Pull request closed without merging**